### PR TITLE
libcatalyst: add package

### PR DESCRIPTION
**Pull request by mathstuf**
There are no official releases yet, so just use a git snapshot for now.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libcatalyst/package.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        def variant_bool(feature, on='ON', off='OFF'):
+            """Ternary for spec variant to ON/OFF string"""
+            if feature in spec:
+                return on
+            return off
+
+        args = [
+            '-DCATALYST_BUILD_TESTING=OFF',
+            '-DCATALYST_USE_MPI=%s' % variant_bool('+mpi'),
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libcatalyst/package.py]: I think you can:
```suggestion
            self.define_from_variant('CATALYST_USE_MPI', 'mpi')
```
and remove the definition of `variant_bool`.

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request

**Comment by mathstuf** [comment]
Nifty, thanks.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks @mathstuf

**Pull request merged**